api/http/handler/endpoints/endpoint_create.go

```diff
@@ -192,7 +192,13 @@ func (handler *Handler) createAzureEndpoint(payload *endpointCreatePayload) (*po
 func (handler *Handler) createUnsecuredEndpoint(payload *endpointCreatePayload) (*portainer.Endpoint, *httperror.HandlerError) {
 	endpointType := portainer.DockerEnvironment
 
-	if !strings.HasPrefix(payload.URL, "unix://") {
+	if payload.Name == "local" && payload.URL == "" {
```

Comment:
No need to check if the name equals local here, we should default to local endpoint if the URL is not specified. Please you update the description of the URL field in `swagger.yml` file as well.